### PR TITLE
4.x: Helidon Arquillian module should only depend on MP core

### PR DESCRIPTION
**Pull request by jbescos**
### Description
Issue https://github.com/helidon-io/helidon/issues/6005

### Documentation
None

### CODE REVIEW

File: microprofile/tests/arquillian/pom.xml [hunk 1]
```diff
@@ -70,5 +64,9 @@
             <groupId>org.testng</groupId>
             <artifactId>testng</artifactId>
         </dependency>
+        <dependency>
```
Comment (tomas-langer) [inline microprofile/tests/arquillian/pom.xml:67]: Why does it depend on MP metrics?

File: microprofile/tests/tck/tck-cdi/pom.xml [hunk 2]
```diff
@@ -28,6 +28,16 @@
     <name>Helidon Microprofile Tests TCK CDI</name>
 
     <dependencies>
+        <dependency>
+            <groupId>io.helidon.microprofile.metrics</groupId>
+            <artifactId>helidon-microprofile-metrics</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (tomas-langer) [inline microprofile/tests/tck/tck-cdi/pom.xml:36]: This is part of helidon MP core, no need to add it

File: microprofile/tests/tck/tck-config/pom.xml [hunk 3]
```diff
@@ -36,6 +36,11 @@
             <artifactId>helidon-microprofile-config</artifactId>
             <scope>test</scope>
         </dependency>
+        <dependency>
```
Comment (tomas-langer) [inline microprofile/tests/tck/tck-config/pom.xml:39]: Why is this here for config TCK? AFAIK metrics are not used from config

File: microprofile/tests/tck/tck-core-profile/tck-core-profile-test/pom.xml [hunk 4]
```diff
@@ -30,6 +30,11 @@
     <name>Helidon Microprofile Tests TCK Core Profile</name>
 
     <dependencies>
+        <dependency>
```
Comment (tomas-langer) [inline microprofile/tests/tck/tck-core-profile/tck-core-profile-test/pom.xml:33]: why do you depend on the whole thing here? It should be sufficient to depend on the core, as core profile should be already supported by core.

File: microprofile/tests/tck/tck-health/pom.xml [hunk 5]
```diff
@@ -31,6 +31,16 @@
     <name>Helidon Microprofile Tests TCK Health</name>
 
     <dependencies>
+        <dependency>
+            <groupId>io.helidon.microprofile.health</groupId>
+            <artifactId>helidon-microprofile-health</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
```
Comment (tomas-langer) [inline microprofile/tests/tck/tck-health/pom.xml:39]: Metrics seem to be added to all poms, including the arqulilian. Please explain

File: microprofile/tests/tck/tck-inject/tck-inject-test/pom.xml [hunk 6]
```diff
@@ -41,8 +41,8 @@
             <scope>test</scope>
         </dependency>
         <dependency>
-            <groupId>io.helidon.microprofile.bundles</groupId>
-            <artifactId>helidon-microprofile</artifactId>
+            <groupId>io.helidon.microprofile.cdi</groupId>
```
Comment (tomas-langer) [inline microprofile/tests/tck/tck-inject/tck-inject-test/pom.xml:44]: If this module depends on the arquillian, then this should not be needed, otherwise this is OK

**Review by tomas-langer — CHANGES_REQUESTED** [review]

File: microprofile/tests/arquillian/pom.xml (hunk 1 above)
Comment (jbescos) [reply microprofile/tests/arquillian/pom.xml:67]: Because of `HelidonDeployableContainer#cleanupBaseMetrics`

File: microprofile/tests/tck/tck-cdi/pom.xml (hunk 2 above)
Comment (jbescos) [reply microprofile/tests/tck/tck-cdi/pom.xml:36]: There are test failures:
```
[ERROR] Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 1.172 s <<< FAILURE! - in org.jboss.cdi.tck.tests.definition.stereotype.inheritance.StereotypeInheritenceTest
[ERROR] org.jboss.cdi.tck.tests.definition.stereotype.inheritance.StereotypeInheritenceTest.arquillianAfterClass  Time elapsed: 0.021 s  <<< FAILURE!
org.jboss.weld.exceptions.UnsatisfiedResolutionException: WELD-001334: Unsatisfied dependencies for type MetricRegistry with qualifiers @RegistryType 
	at org.jboss.weld.bean.builtin.InstanceImpl.checkBeanResolved(InstanceImpl.java:251)
	at org.jboss.weld.bean.builtin.InstanceImpl.get(InstanceImpl.java:113)
```

mvn dependency:tree does not show `helidon-microprofile-metrics`

File: microprofile/tests/tck/tck-inject/tck-inject-test/pom.xml (hunk 6 above)
Comment (jbescos) [reply microprofile/tests/tck/tck-inject/tck-inject-test/pom.xml:44]: It does not depend on Arquillian. What dependency should we add here?. Tests are located under microprofile, so `io.helidon.microprofile.cdi` seems correct to me.

File: microprofile/tests/tck/tck-health/pom.xml (hunk 5 above)
Comment (jbescos) [reply microprofile/tests/tck/tck-health/pom.xml:39]: See the first comments. It is required if you use Helidon Arquillian.

File: microprofile/tests/tck/tck-config/pom.xml (hunk 3 above)
Comment (jbescos) [reply microprofile/tests/tck/tck-config/pom.xml:39]: As long as you have this dependency, you need metrics:
```
        <dependency>
            <groupId>io.helidon.microprofile.tests</groupId>
            <artifactId>helidon-arquillian</artifactId>
        </dependency>

```
We could add it directly in Helidon-Arquillian but I wanted to remove any microprofile dependency from there.

File: microprofile/tests/tck/tck-core-profile/tck-core-profile-test/pom.xml (hunk 4 above)
Comment (jbescos) [reply microprofile/tests/tck/tck-core-profile/tck-core-profile-test/pom.xml:33]: Right, for some reason I thought this was testing a small parts of different microprofile things.

**Comment by jbescos** [comment]
I have updated helidon-arquillian to not require metrics in every TCK

**Review by tomas-langer — APPROVED** [review]
